Contact/ContactRepository.cs

```diff
@@ -42,27 +46,44 @@ public Contact Get(ContactRepository database, int id)
         }
 
         //Delete - set a flag
-        public  void Delete(ContactRepository database, int id)
+        public void Delete(ContactRepository database, int id)
         {
             database.ContactList.Find(i => i.contactID.Equals(id)).isActiveRecord = false;
         }
 
         //Search all fields for matches
-        public  List<Contact> Filter(ContactRepository database, string parameter)
+        public List<Contact> Filter(ContactRepository database, string parameter)
         {
-            var results =
-                from contact in database.Get(database)
-                where contact.isActiveRecord && (
-                contact.firstName.Contains(parameter) ||
-                contact.lastName.Contains(parameter) ||
-                contact.email.Contains(parameter) ||
-                contact.CellPhone.Contains(parameter) ||
-                contact.HomePhone.Contains(parameter) ||
-                contact.WorkPhone.Contains(parameter))
-                select contact;
+            var results = new List<Contact>();
 
-            return results.ToList();
+            if (parameter.Contains(lastSearchString) && lastSearchString != string.Empty)
+                if(parameter == lastSearchString)
+                    results = lastSearch;
+                else
+                    results = searchAllFields(lastSearch, parameter);
+            else   
+                results = searchAllFields(database.ContactList, parameter);
+
+            lastSearch = results;
+            lastSearchString = parameter;
+
+            return results;
         }
 
+        private List<Contact> searchAllFields(List<Contact> database, string parameter)
```

Comment:
would be nice to search on multiple fields

Reply:
My logic has been to create a single search field which will search all fields. I suppose I should create another version which would function more as a field-specific search function. I'll put this on my to-do list.

Reply:
Not a big deal.  A typical search might be something like "Brian Allan" in which your way wouldn't work.  I would punt on this as there is plenty more to do.